plugins/action_generate_name/src/generateName.ts

```diff
@@ -0,0 +1,102 @@
+import { ActionQueueStatus } from '../../../src/generated/graphql'
+import { ActionPluginOutput, ActionPluginInput } from '../../types'
+import evaluateExpression from '@openmsupply/expression-evaluator'
+import { EvaluatorNode } from '@openmsupply/expression-evaluator/lib/types'
+import replaceAsync from 'string-replace-async'
+import modifyRecord from '../../action_modify_record/src/modifyRecord'
+
+async function generateName({
+  parameters,
+  applicationData,
+  DBConnect,
+}: ActionPluginInput): Promise<ActionPluginOutput> {
+  const {
+    tableName = 'application',
+    fieldName = 'name',
+    matchField = 'id',
+    matchValue = applicationData?.applicationId,
+    shouldUpdateDatabase = true,
+    additionalData = {},
+    fallbackString = 'UNRESOLVED',
+    formatExpression = '${applicationData.templateName} — ${applicationData.applicationSerial}',
+  } = parameters
+
+  try {
+    const generatedString = await generateString(
+      formatExpression,
+      {
+        applicationData,
```

Comment:
Why not `...applicationData` so in the formatted string you don't have to specify `applicationData` on each field?

Reply:
This is a good point, and I tried it both ways (and it's the same issue for the generateTextString, hence it's still relevant here). Since we can add any amount of additional data to this object, I thought it would be cleaner (and maybe safer) to keep "applicationData" encapsulated in its own object rather than mixing it up with all the other properties -- at the expense of more verbose expressions. (and this is the way we do it for evaluator expressions) But I'm happy to be convinced otherwise -- let's have a quick chat about it after you've looked at #454.